extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/TenantPolicyConfigResolver.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.keycloak.pep;
+
+import java.util.function.Supplier;
+
+import io.quarkus.keycloak.pep.runtime.KeycloakPolicyEnforcerTenantConfig;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * A tenant resolver is responsible for resolving the {@link KeycloakPolicyEnforcerTenantConfig} for tenants, dynamically.
+ */
+public interface TenantPolicyConfigResolver {
+
+    /**
+     * Returns a {@link KeycloakPolicyEnforcerTenantConfig} given a {@code RoutingContext} and tenant id.
+     *
+     * @param routingContext the routing context
+     * @param tenantId tenant id
+     * @param requestContext request context
+     *
+     * @return the tenant configuration. If the uni resolves to {@code null}, indicates that the default
+     *         configuration/tenant should be chosen
+     */
+    Uni<KeycloakPolicyEnforcerTenantConfig> resolve(RoutingContext routingContext, String tenantId,
+            KeycloakRequestContext requestContext);
+
+    /**
+     * Keycloak Context that can be used to run blocking tasks.
```

Comment:
pfff :-D good question, but I simply thought it would be weird to use `OidcRequestContext` inside Keycloak extension. It's different place. If you are sure that `OidcRequestContext` will also be relevant for Kecloak and you want add there non-Keycloak stuff in the future, let's reuse it. I'd love to drop another class, but I am not sure it is a smart move in regards of future breaking changes.

Reply:
I'll use `OidcRequestContext` then.

Reply:
Thanks